src/worker.cpp

```diff
@@ -1193,4 +1225,301 @@ void Worker::collectFrusFromJson()
+            * std::tuple<types::Record, types::Keyword>(l_dstRecordName,
+            l_dstKeyword)); l_dstStrValue =
+            std::string(l_dstBinaryValue.begin(),
+            l_dstBinaryValue.end());
+            */
+        }
+
+        if (l_sourceBinaryValue != l_dstBinaryValue)
+        {
+            if (l_dstBinaryValue == l_defaultBinaryValue)
+            {
+                // TODO: Uncomment when writeKeyword API implemenation goes
+                // in, handle if separate update to hardware and cache is
+                // required. utils::writeKeyword(i_dstPath,
+                // types::IpzData(l_dstRecordName, l_dstKeyword
+                // l_sourceBinaryValue));
+                io_dstVpdMap[l_dstRecordName][l_dstKeyword] = l_sourceStrValue;
+                continue;
+            }
+
+            // Update map to publish the same data on DBus, which is already
+            // present on the DBus.
+            if (l_isEmptyDstVpdReceived)
+            {
+                io_sourceVpdMap[l_sourceRecordName][l_sourceKeyword] =
+                    l_dstStrValue;
+            }
+
+            if (l_sourceBinaryValue == l_defaultBinaryValue)
+            {
+                // TODO: Uncomment when writeKeyword API implemenation goes
+                // in, handle if separate update to hardware and cache is
+                // required. utils::writeKeyword(i_sourcePath,
+                // types::IpzData(l_sourceRecordName, l_sourceKeyword,
+                // l_dstBinaryValue));
+            }
+            else
+            {
+                // Uncomment when PEL implementation goes in.
+                /*inventory::PelAdditionalData additionalData{};
+                std::string errMsg = "Mismatch found between source and
+                destination VPD for record: " + l_sourceRecordName + " and
+                keyword: " + l_sourceKeyword;
+                additionalData.emplace("DESCRIPTION", errMsg);
+                additionalData.emplace("Value read from destination: ",
+                l_dstStrValue); additionalData.emplace("Value read from
+                source: ", l_sourceStrValue); createPEL(additionalData,
+                PelSeverity::WARNING, errIntfForVPDMismatch, nullptr);
+                */
+            }
+        }
+        else if (l_sourceBinaryValue == l_defaultBinaryValue &&
+                 l_dstBinaryValue == l_defaultBinaryValue && l_isPelRequired)
+        {
+            logging::logMessage(
+                "Default value found in both source and destination VPD, for record: " +
+                l_sourceRecordName + " and keyword: " + l_sourceKeyword);
+        }
+    }
+}
+
+bool Worker::isBackupAndRestoreRequired() const
+{
+    return m_performBackupAndRestore;
+}
+
+std::string Worker::getBackupRestoreJsonPath() const
+{
+    std::string l_backupRestoreJsonPath;
+    if (auto l_targetConfigPath =
+            read_symlink(std::filesystem::path(INVENTORY_JSON_SYM_LINK));
```

Comment:
should we keep IM keyvalue as worker class member and relay on IM keyvalue directly ?

Reply:
IM is specific to IBM.
Even the symlink part is specific. Need to give some more thoughts here.